ihatemoney/forms.py

```diff
@@ -131,9 +132,11 @@ def update(self, project):
         """Update the project with the information from the form"""
         project.name = self.name.data
 
-        # Only update password if changed to prevent spurious log entries
-        if not check_password_hash(project.password, self.password.data):
-            project.password = generate_password_hash(self.password.data)
+        if self.password.data:
+            # Only update password if different from the previous one,
+            # to prevent spurious log entries
+            if not check_password_hash(project.password, self.password.data):
```

Comment:
Those two conditions should be merged into one I guess

Reply:
```suggestion
        # Only update password if different from the previous one,
        # to prevent spurious log entries
        if self.password.data and not check_password_hash(project.password, self.password.data):
```

Reply:
Right, it was kind of intended because the comment only applies to the second test, but I will improve it then :)

Reply:
Let me know what you think of the new version.

Reply:
Nice integration of comments!